relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -6,40 +6,56 @@ use actix::prelude::*;
 use futures::{future, prelude::*};
 
 use relay_general::protocol::{Event, EventId};
-use relay_sampling::{get_matching_event_rule, pseudo_random_from_uuid, rule_type_for_event};
+use relay_sampling::{
+    get_matching_event_rule, pseudo_random_from_uuid, rule_type_for_event, RuleId,
+};
 
 use crate::actors::project::{GetCachedProjectState, GetProjectState, Project, ProjectState};
 use crate::envelope::{Envelope, ItemType};
 
+/// The result of a Sampling operation
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum SamplingResult {
+    /// Keep event
+    Keep,
+    /// Drop event, due to rule with provided Id
+    Drop(RuleId),
+    /// No decision made  
+    NoDecision,
+}
+
 // Checks whether an event should be kept or removed by dynamic sampling
 pub fn should_keep_event(
     event: &Event,
     ip_addr: Option<IpAddr>,
     project_state: &ProjectState,
     processing_enabled: bool,
-) -> Option<bool> {
+) -> SamplingResult {
     let sampling_config = match &project_state.config.dynamic_sampling {
-        None => return None, // without config there is not enough info to make up my mind
+        None => return SamplingResult::NoDecision, // without config there is not enough info to make up my mind
         Some(config) => config,
     };
 
     // when we have unsupported rules disable sampling for non processing relays
     if !processing_enabled && sampling_config.has_unsupported_rules() {
-        return Some(true);
+        return SamplingResult::Keep;
     }
 
     let event_id = match event.id.0 {
-        None => return None, // if no eventID we can't really sample so keep everything
+        None => return SamplingResult::NoDecision, // if no eventID we can't really sample so keep everything
```

Comment:
This comment seems out of sync. You're returning `NoDecision`, but the comment says keep. 

This is another indicator for me that the `NoDecision` enum variant may not be appropriate here. In fact, we're making the decision to explicitly keep an event, so the code should also reflect that.

Also, if we use the word "sample" in our comments, let's use the following interpretation:
- sample: keep (as in "take a sample")
- not sample: drop